generator/src/main/java/com/scottlogic/deg/generator/decisiontree/MostProlificConstraintOptimiser.java

```diff
@@ -143,12 +121,10 @@ private int disfavourNotConstraints(Map.Entry<AtomicConstraint, List<AtomicConst
 
     private AtomicConstraint getMostProlificAtomicConstraint(Collection<DecisionNode> decisions) {
         Map<AtomicConstraint, List<AtomicConstraint>> decisionConstraints =
-            FlatMappingSpliterator.flatMap(
-                FlatMappingSpliterator.flatMap(
-                    decisions.stream(),
-                    dn -> dn.getOptions().stream()),
-                option -> option.getAtomicConstraints().stream())
-            .collect(Collectors.groupingBy(Function.identity()));
+                decisions.stream()
+                    .flatMap(dn -> dn.getOptions().stream())
```

Comment:
Is the flatmap eagerly evaluating issue outlined in the `FlatMappingSpliterator` docs an issue here?

Reply:
nope, as the whole stream will be collected